core/src/test/java/org/apache/iceberg/TestPartitionSpecInfo.java

```diff
@@ -95,6 +95,19 @@ public void testSpecInfoPartitionedTable() {
         .doesNotContainKey(Integer.MAX_VALUE);
   }
 
+  @TestTemplate
+  public void testSpecInfoPartitionedTableCaseInsensitive() {
+    PartitionSpec spec =
+        PartitionSpec.builderFor(schema).caseSensitive(false).identity("DATA").build();
```

Comment:
can you also add a test case where `caseSensitive(true)` and spec cannot be built?

Reply:
test case has been added.